aiida_quantumespresso/workflows/pw/bands.py

```diff
@@ -217,7 +217,7 @@ def run_bands(self):
         if 'nbands_factor' in self.inputs:
             factor = self.inputs.nbands_factor.value
             parameters = self.ctx.workchain_scf.outputs.output_parameters.get_dict()
-            nspin = int(parameters['number_of_spin_components'])
+            nspin = min(2, int(parameters['number_of_spin_components']))
```

Comment:
Although this fixes it, I think this is a bit cryptic.
Could we instead make it:
```python
if int(parameters['number_of_spin_components']) > 1:
    nspin_factor = 2
else:
    nspin_factor = 1
....
nbnd = max(int(0.5 * nelectron * nspin_factor * factor), int(0.5 * nelectron * nspin_factor) + 4 * nspin_factor, nbands)
```

Reply:
This would indeed improve the readability of the code.
I have added the proposed changes.